### PR TITLE
feat: Add V2 content generation with streaming UI, AI SDK gateway upgrade, and breaking config changes

### PR DESCRIPTION
**Pull request by toyamarinyon**
## Overview

This PR introduces a major upgrade to the content generation system, migrating to AI SDK v2 and implementing a new streaming-based execution engine. The changes enable real-time generation with cancellation support, improved tool integration, and better provider-specific handling for OpenAI, Anthropic, and Google models.

### Breaking Changes ⚠️
- **Language model configuration updates:**
  - Anthropic: `reasoningText` renamed to `thinking`
  - Google Gemini 2.5: `urlContext` removed, `thinking` option added with new defaults
- **AI SDK gateway upgraded to v2** with significant version bumps across AI SDK libraries
- **Configuration validation:** `parseConfiguration` now validates against Zod schemas, falling back to defaults for invalid values

### Key Improvements
- **Enhanced UI for Content Generation nodes:**
  - Direct run/stop controls in properties panel
  - Real-time streaming output with cancellation support
  - Improved layout with split Prompt/Output columns

- **New V2 execution engine:**
  - AI SDK `streamText` pipeline integration
  - Provider-specific configuration mapping (OpenAI, Anthropic, Google)
  - Expanded tool system with GitHub, Postgres, and web search
  - Improved message handling with file/web page attachments
  - Robust streaming lifecycle with abort, chunk batching, and usage tracking

- **Language model registry enhancements:**
  - Added Google Gemini 3 Pro Preview model
  - Zod schema validation for configurations and tool inputs
  - New helper functions for tool management
  - Comprehensive test coverage

## Testing

- Unit tests added for `parseConfiguration` utility
- Test scripts included for language model registry
- Verified streaming generation with cancellation across providers
- Validated tool integration with secrets resolution
- Tested configuration migration paths for breaking changes

## Review Notes

- **Breaking changes impact:** Please review the configuration key changes and ensure documentation is updated
- **V2 migration path:** The new execution engine runs alongside the existing one - feedback needed on transition strategy
- **Tool system architecture:** New tool wiring approach uses Zod schemas - looking for input on extensibility
- **Performance considerations:** Chunk batching to storage may need tuning based on usage patterns

## Related Issues

This PR lays the foundation for improved content generation capabilities and addresses the need for better streaming support and provider-specific optimizations.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- **Major AI SDK v2 upgrade:** Migrated to AI SDK v2 gateway (1.0.23 → 2.0.15) with updated provider packages for Anthropic, Google, and OpenAI

- **V2 content generation engine:** New streaming-based execution engine with real-time generation, cancellation support, and chunk batching

- **Enhanced UI for content generation:** Added run/stop controls, real-time streaming output display, and split prompt/output layout in properties panel

- **Breaking configuration changes:**
  - Anthropic: `reasoningText` renamed to `thinking`
  - Google Gemini 2.5: `urlContext` removed, `thinking` option added with new defaults
  - Google Gemini 3 Pro Preview model added

- **Expanded tool system:** Implemented GitHub and Postgres tools with Zod schema validation, provider-specific web search integration (Anthropic, Google, OpenAI)

- **Configuration validation:** New `parseConfiguration` utility validates unknown config data against Zod schemas with fallback to defaults

- **Language model transformation:** Provider-specific configuration mapping for OpenAI reasoning, Anthropic thinking, and Google thinking levels

- **Comprehensive test coverage:** Unit tests for `parseConfiguration` utility and language model registry

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["AI SDK v2<br/>Gateway Upgrade"] --> B["V2 Execution<br/>Engine"]
  B --> C["Streaming<br/>Generation"]
  C --> D["Content Generation<br/>UI"]
  B --> E["Tool System"]
  E --> F["GitHub<br/>Tools"]
  E --> G["Postgres<br/>Tools"]
  E --> H["Web Search<br/>Tools"]
  I["Config<br/>Validation"] --> B
  J["Language Model<br/>Transform"] --> B
  K["Breaking Changes:<br/>Anthropic/Google"] --> I
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>18 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>github.ts</strong><dd><code>GitHub tools implementation for V2 streaming execution engine</code></dd></summary>
<hr>

packages/giselle/src/generations/v2/tools/github.ts

<ul><li>New file implementing GitHub tool integration for V2 content <br>generation engine<br> <li> Exports <code>createGitHubTools</code> function that maps tool definitions to AI <br>SDK tool implementations<br> <li> Implements 40+ GitHub API operations (issues, PRs, branches, commits, <br>search, etc.)<br> <li> Each tool uses Octokit client for API calls with proper parameter <br>mapping and error handling</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-d70cce79a431a585c692669ecbfb00d487af716b0d265c39fc4f361cfd727c7f">+832/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>github.ts</strong><dd><code>GitHub tools schema validation and type safety improvements</code></dd></summary>
<hr>

packages/language-model-registry/src/tools/github.ts

<ul><li>Adds Zod schemas to all 40 GitHub tool definitions for input <br>validation<br> <li> Renames array from <code>tools</code> to <code>githubTools</code> and exports <code>GitHubTool</code> type<br> <li> Adds <code>isGitHubToolName</code> type guard function for runtime validation<br> <li> Each tool now includes detailed schema with field descriptions and <br>constraints</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-29d3ecd5efd032b4c244d44869f998442749b60c4fa7ac216e7ba11ee18e9ba1">+420/-2</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generate-content.ts</strong><dd><code>V2 content generation execution engine with streaming support</code></dd></summary>
<hr>

packages/giselle/src/generations/generate-content.ts

<ul><li>Adds new <code>generateContentV2</code> function for V2 content generation <br>execution<br> <li> Imports V2-specific utilities for language model transformation and <br>tool building<br> <li> Routes <code>ContentGenerationNode</code> types to V2 engine while keeping V1 for <br><code>TextGenerationNode</code><br> <li> Implements streaming with chunk batching, abort handling, and <br>comprehensive error management</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-d1c7985f77561f9bd924107a6e24bb1cb06973dd20a0b61f432439c9405106b1">+277/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>utils.ts</strong><dd><code>Content generation message building with attachment support</code></dd></summary>
<hr>

packages/giselle/src/generations/utils.ts

<ul><li>Separates <code>buildGenerationMessageForContentGeneration</code> from text <br>generation message building<br> <li> Implements content generation message building with support for file <br>attachments and web pages<br> <li> Handles provider-specific attachment strategies (Anthropic/OpenAI vs <br>Google)<br> <li> Processes template keywords for dynamic content injection from context <br>nodes</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-7eddd355f330c712d951b4b57360a25e80ebbd4e05aca548189f6c8303e62e18">+225/-2</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>build-tool-set.ts</strong><dd><code>V2 tool set builder with provider integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/src/generations/v2/tools/build-tool-set.ts

<ul><li>New file implementing tool set builder for V2 content generation<br> <li> Handles tool configuration validation and secret resolution for GitHub <br>and Postgres<br> <li> Integrates provider-specific tools (Anthropic web search, Google <br>search, OpenAI web search)<br> <li> Routes tool creation based on language model tool type with proper <br>error handling</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-3cccc74fa184218150295a2824445294816c7717293956de9810b7b95a2d472f">+178/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>tool.ts</strong><dd><code>Tool type system enhancement with schema support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/tool.ts

<ul><li>Adds optional <code>schema</code> field to <code>BaseToolConfigurationOption</code> for Zod <br>validation<br> <li> Splits <code>Tool</code> type into <code>ToolWithoutSchema</code> and <code>ToolWithSchema</code> variants<br> <li> Adds <code>hasSchema</code> type guard function for runtime schema detection<br> <li> Refactors <code>LanguageModelTool</code> into variants with/without tools and adds <br>helper functions</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-3af907fbd24f45919b0812358e56ae383a40f4727ad894647c2c9e0b3b1ec070">+116/-5</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>google.ts</strong><dd><code>Google Gemini model updates with thinking configuration</code>&nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/google.ts

<ul><li>Adds new <code>google/gemini-3-pro-preview</code> model with thinking and grounding <br>support<br> <li> Renames <code>urlContext</code> to <code>thinking</code> for Gemini 2.5 models with new defaults<br> <li> Updates configuration schemas with <code>thinkingLevel</code> enum and <br><code>searchGrounding</code> boolean<br> <li> Adjusts default configurations across all Google models for <br>consistency</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-0d83151a4b2c2224cd1c5be9b8fb546ebd126972bb61ea6615828f2cee54537d">+56/-12</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>transform-giselle-to-ai-sdk.ts</strong><dd><code>Language model configuration transformation for AI SDK</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/src/generations/v2/language-model/transform-giselle-to-ai-sdk.ts

<ul><li>New file implementing transformation of Giselle language model config <br>to AI SDK format<br> <li> Maps OpenAI reasoning effort and text verbosity settings<br> <li> Handles Anthropic thinking configuration with token budgets<br> <li> Converts Google thinking levels and dynamic thinking budgets based on <br>model type</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-0a09de5e2c8b926e5c6fe97190e30c07afcc824bdad76066f6d54272a544c12f">+103/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>language-model.ts</strong><dd><code>Configuration parsing utility with Zod validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/language-model.ts

<ul><li>Adds new <code>parseConfiguration</code> function for validating unknown config <br>data against model schemas<br> <li> Validates each configuration option using Zod schemas with fallback to <br>defaults<br> <li> Provides comprehensive documentation with usage examples<br> <li> Handles missing values, type mismatches, and range violations <br>gracefully</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-b82b17fc5122b0353e0434eb69257d71aafcde52c6c2fd4facff3e211bc5d373">+55/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>postgres.ts</strong><dd><code>Postgres tools implementation for V2 execution engine</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/src/generations/v2/tools/postgres.ts

<ul><li>New file implementing Postgres tool integration for V2 content <br>generation<br> <li> Exports <code>createPostgresTool</code> function that creates database query tools<br> <li> Implements <code>getTableStructure</code> and <code>query</code> tools with connection pooling<br> <li> Includes proper resource cleanup via context waitUntil callback</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-4775739a4d2c782fd79c0c67f28daead6df5ae0ae9d64d3897296adc1bb19cef">+72/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>anthropic-web-search.ts</strong><dd><code>Anthropic web search tool schema validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/anthropic-web-search.ts

<ul><li>Adds Zod schemas to configuration options for <code>maxUses</code>, <code>allowedDomains</code>, <br>and <code>blockedDomains</code><br> <li> Schemas define optional arrays and numbers with proper type validation</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-72e12879bdebdec920e82f34bdb6f5f546d145ddcb8d492e87ac2f5f053346b4">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>postgres.ts</strong><dd><code>Postgres tools schema validation and type exports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/postgres.ts

<ul><li>Renames <code>tools</code> array to <code>postgresTools</code> and exports <code>PostgresTool</code> type<br> <li> Adds Zod schema to <code>query</code> tool with string constraints (min 1, max 1000 <br>chars)<br> <li> Maintains <code>getTableStructure</code> tool without schema requirements</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-8659bbfca590722a1806c5276d4006f51fe52bcb28bbec37f6114221aa7f7641">+7/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>openai-web-search.ts</strong><dd><code>OpenAI web search tool schema validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/openai-web-search.ts

<ul><li>Adds Zod schema to <code>allowedDomains</code> configuration option<br> <li> Schema defines optional array of strings for domain filtering</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-dea3b111f2eeb34e9c6499b046f9afa47ed90daae751a13ed84a68fe79de11d9">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>tools.ts</strong><dd><code>Tool registry lookup utility function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/tools.ts

<ul><li>Adds new <code>getLanguageModelTool</code> function to retrieve tool by name with <br>error handling<br> <li> Throws error if tool name is not found in registry</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-cf3671d339b69de221b03d748bed0a3fc7a843323730a7e99d10662544a2e3ac">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Octokit type export for tool integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/github-tool/src/index.ts

- Exports `Octokit` type from `@octokit/core` for external use

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-df952beb9628e7ae8d92a420308bf736117aa91930334d593c8c061d2b9c3f28">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Content generation node type export</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/protocol/src/node/operations/index.ts

- Adds export for `content-generation` operation node type

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-ab7a5f80e77e2b8b5af6e00dec0f003f40f9367fe27da2b7fedaace1e258a363">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>content-generation-node-properties-panel.tsx</strong><dd><code>Content generation UI with streaming and run controls</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal-packages/workflow-designer-ui/src/editor/properties-panel/content-generation-node-properties-panel/content-generation-node-properties-panel.tsx

<ul><li>Adds generation runner integration with <code>useNodeGenerations</code> hook<br> <li> Implements run/stop buttons in properties panel with generation state <br>management<br> <li> Splits prompt/output layout into 50/50 columns with real-time <br>generation display<br> <li> Adds <code>GenerationView</code> component to display streaming generation results</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-b37464f40d374545b1bd956acf7fbf08c71fd180d6a0f1f5fbd3e4799239d32c">+70/-21</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>generation-runner.tsx</strong><dd><code>Generation runner support for content generation nodes</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/react/src/generations/generation-runner.tsx

<ul><li>Updates type guard to accept both <code>TextGenerationNode</code> and <br><code>ContentGenerationNode</code><br> <li> Imports <code>isContentGenerationNode</code> type guard function</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-5d1f65056c99c46a1cabf8cc278d0bc38ec2b4495672d9d0970e2713732b3629">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>language-model.test.ts</strong><dd><code>Configuration parsing unit tests for language models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/language-model.test.ts

<ul><li>New test file with comprehensive test coverage for <code>parseConfiguration</code> <br>utility<br> <li> Tests valid configuration parsing, fallback to defaults, type <br>validation, and edge cases<br> <li> Covers Anthropic, OpenAI, and Google models with various configuration <br>scenarios<br> <li> Validates enum values, range constraints, and missing/invalid data <br>handling</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-0f8da27cccf51e74f471924fa743d00b607f29c71fbec1d907dd9c893b58ff2b">+168/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>anthropic.ts</strong><dd><code>Anthropic model configuration key rename (breaking change)</code></dd></summary>
<hr>

packages/language-model-registry/src/anthropic.ts

<ul><li>Renames <code>reasoningText</code> configuration option to <code>thinking</code> across all <br>Anthropic models<br> <li> Updates default configuration values to use <code>thinking: false</code><br> <li> Maintains schema and description consistency for the renamed option</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-a0b11c3767335b5cb83b61542e8bf5e67994adf81844bd1c0eeebaf7b41bec12">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>V2 tools module exports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/src/generations/v2/tools/index.ts

<ul><li>New barrel export file for V2 tools module<br> <li> Exports <code>buildToolSet</code>, <code>createGitHubTools</code>, and <code>createPostgresTool</code></ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-b7ae9ba76982f3e25624efc2bd13bdd5e16ebb20b859158960a20a22742c6312">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Tool registry exports update</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/src/tools/index.ts

- Adds exports for GitHub and Postgres tool definitions

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-0a995da4f6629202e250d23580d6965305c56b6cf95cd41000c6cb971ef2c3fb">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>V2 language model module exports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/src/generations/v2/language-model/index.ts

<ul><li>New barrel export file for V2 language model module<br> <li> Exports language model transformation utilities</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-6b011b0382e8e6df17ddaf9c303adba4312457d75d4833a068dfa5084c6e377c">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>packages-license.md</strong><dd><code>License documentation and dependency version updates</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/packages-license.md

<ul><li>Updates AI SDK dependency versions (anthropic, google, openai, <br>gateway)<br> <li> Updates <code>ai</code> package version from 5.0.51 to 5.0.101<br> <li> Removes obsolete platform-specific dependencies from license summary</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-ae1da9a37e22d3531ffedff27e1dd92e0c755e79f6062620c4370f8dde00b2ab">+9/-31</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pnpm-workspace.yaml</strong><dd><code>AI SDK dependency version upgrades</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pnpm-workspace.yaml

<ul><li>Updates <code>@ai-sdk/anthropic</code> from 2.0.17 to 2.0.49<br> <li> Updates <code>@ai-sdk/gateway</code> from 1.0.23 to 2.0.15 (major version bump)<br> <li> Updates <code>@ai-sdk/google</code> from 2.0.14 to 2.0.43<br> <li> Updates <code>@ai-sdk/openai</code> from 2.0.30 to 2.0.72<br> <li> Updates <code>ai</code> from 5.0.51 to 5.0.101</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-18ae0a0fab29a7db7aded913fd05f30a2c8f6c104fadae86c9d217091709794c">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Language model registry dependency addition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/giselle/package.json

<ul><li>Adds <code>@giselles-ai/language-model-registry</code> as dependency for V2 content <br>generation</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-df9e5121325c97dac56dc52dc9035438ff868d9800dc4cfd633b6fbef19255c3">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>pnpm-lock.yaml</strong><dd><code>AI SDK v2 gateway upgrade with provider package updates</code>&nbsp; &nbsp; </dd></summary>
<hr>

pnpm-lock.yaml

<ul><li>Upgraded AI SDK packages to latest versions: <code>@ai-sdk/anthropic</code> (2.0.17 <br>→ 2.0.49), <code>@ai-sdk/openai</code> (2.0.30 → 2.0.72), <code>@ai-sdk/google</code> (2.0.14 → <br>2.0.43), and <code>@ai-sdk/gateway</code> (1.0.23 → 2.0.15)<br> <li> Updated core <code>ai</code> package from 5.0.51 to 5.0.101 across multiple <br>workspace packages<br> <li> Updated <code>@ai-sdk/provider-utils</code> from 3.0.9 to 3.0.17 to support new <br>provider implementations<br> <li> Added <code>@vercel/oidc@3.0.5</code> as a new dependency for <br><code>@ai-sdk/gateway@2.0.15</code><br> <li> Updated OpenTelemetry dependencies and resolved webpack/esbuild <br>version conflicts across build configurations</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+183/-99</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Test infrastructure setup for language model registry</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/language-model-registry/package.json

<ul><li>Adds <code>test</code> script for running vitest<br> <li> Adds <code>vitest</code> as dev dependency</ul>

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2266/files#diff-816e581ab550e077dcf9ca4e82f1d515e52a77d2101faf8b873b336150a4bc0a">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

* **New Features**
  * Content-generation flow and UI now support a v2 generation pipeline with streaming output and workspace-aware runner.
  * New tool integrations: GitHub automation, PostgreSQL query/table tools, and web search tooling.

* **Improvements**
  * Expanded language-model registry: new Google Gemini preview, "thinking" options, richer provider mappings, and safer configuration parsing.
  * Enhanced validation and schema support for tools and model configurations.

* **Updates**
  * Several AI SDK packages bumped to newer versions; registry package added and test coverage expanded.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

> [!NOTE]
> Adds a streaming V2 content-generation engine with run/stop UI, provider-specific config mapping, and validated GitHub/Postgres/web-search tools; upgrades AI SDK gateway to v2 and updates model configs (breaking).
> 
> - **Content Generation V2 (engine + runtime)**:
>   - Route `contentGeneration` nodes to new streaming pipeline using `ai.streamText`, with abort, batching, usage, and provider metadata.
>   - Build messages with file/webpage attachments; add tool execution via validated `ToolSet`.
> - **UI/Runner**:
>   - Properties panel: run/stop controls, split Prompt/Output, live output via `GenerationView`.
>   - React runner accepts both `textGeneration` and `contentGeneration` nodes.
> - **Tools**:
>   - New builders for GitHub and Postgres tools; provider web search (Anthropic/OpenAI/Google).
>   - Zod-backed input/config schemas; secret resolution; Octokit type export.
> - **Language Model Registry**:
>   - Add `google/gemini-3-pro-preview`.
>   - Breaking config: Anthropic `reasoningText` → `thinking`; Google Gemini 2.5: add `thinking`, remove `urlContext`.
>   - New `parseConfiguration` utility (+ tests) and helpers (`getLanguageModelTool`, schema/type guards).
>   - Transform Giselle configs to AI SDK call options per provider.
> - **Protocol/Utils**:
>   - Export `content-generation` node; separate message builders for content vs text/image.
> - **Dependencies**:
>   - Upgrade AI SDK packages (gateway v2.0.15; anthropic/google/openai) and `ai` to 5.0.101; wire workspace dep for language-model-registry.
> - **Docs**:
>   - Update licenses/versions in `docs/packages-license.md`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7dbd44e135156fe5e863fbe81666efdd554d137b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by shige** [comment]
@codex review

**Comment by shige** [comment]
@coderabbitai review

File: packages/giselle/src/generations/v2/tools/postgres.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+					}),
+					execute: async ({ query }) => {
+						try {
+							const res = await pool.query(query);
+							return res.rows;
+						} catch (e) {
+							return e;
+						}
+					},
+				});
```
[the remark this answers is not in the document]
Comment (shige) [reply packages/giselle/src/generations/v2/tools/postgres.ts:63]: 🍠 IMO:

I thought it might be a good idea to fix it separately, or just temporarily hide the Postgres query tool itself.

File: packages/giselle/src/generations/v2/language-model/transform-giselle-to-ai-sdk.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			return {
+				providerOptions: {
+					anthropic: {
+						thinking: {
+							type: "enabled",
+							budgetTokens: 12000,
+						},
+					} satisfies AnthropicProviderOptions,
+				},
+			} as Partial<LanguageModelV2CallOptions>;
```
[the remark this answers is not in the document]
Comment (shige) [reply packages/giselle/src/generations/v2/language-model/transform-giselle-to-ai-sdk.ts:64]: If this is not the intended implementation, I would appreciate it if you could fix it!

**Review by shige — APPROVED** [review]
Thank you! 🚀

**Comment by toyamarinyon** [comment]
I've fixed some suggestions from review agents 4a4156a4a25409eb411ed47eea9886a06a448db7 428ad3369c4e238a56b670ede9104439ce671d24  261206f3d4952653f3b543450d88c97253dbc410 24c23c1aba0d00e1fbcec74bdba235070a26d870 696f2a38e90307e74b27184282b81cf6ba480064
